### PR TITLE
add telemetry and progressBar

### PR DESCRIPTION
**Pull request by Eskibear**
* telemetry for editStarters steps
* progress bar for metadata fetching

### CODE REVIEW

File: src/Routines.ts [hunk 1]
```diff
@@ -169,37 +190,48 @@ export module Routines {
...
             }
+
             // modify xml object
-            const newXml: any = Object.assign({}, xml);
+            const newXml: { project: XmlNode } = getUpdatedPomXml(xml, starters, toRemove, toAdd);
+
+            // re-generate a pom.xml
+            const output: string = Utils.buildXmlContent(newXml);
+            await fse.writeFile(entry.fsPath, output);
+            vscode.window.showInformationMessage("Successfully updated.");
```
Comment (jdneo) [inline src/Routines.ts]: Make the hint more meaningful? e.g. `xxx has been successfully updated.`

File: src/pomxml/PomXml.ts [hunk 2]
```diff
@@ -1,65 +1,86 @@
...
 // Licensed under the MIT license.
 
-import { IBomNode, IDependencyNode, IRepositoryNode } from "../Interfaces";
+import * as _ from "lodash";
+import { IBomNode, IDependencyNode, IRepositoryNode, XmlNode } from "../Interfaces";
 
-export function addDependencyNode(xmlObj: any, node: IDependencyNode): void {
-    //todo insert if not exists
-    xmlObj.project.dependencies[0].dependency.push(node);
+function isNullOrEmptyNode(node: any): boolean {
```
Comment (jdneo) [inline src/pomxml/PomXml.ts]: `isEmpty()` already contains null check

File: src/Routines.ts (hunk 1 above)
Comment (Eskibear) [reply src/Routines.ts]: so what's `xxx`? project name? pom file path? Cannot figure out a proper one, what do you suggest ?

File: src/pomxml/PomXml.ts (hunk 2 above)
Comment (Eskibear) [reply src/pomxml/PomXml.ts]: got it

**Pull request merged**